tests/test_sparse.py

```diff
@@ -9,11 +9,21 @@
 class TestSparseDIA:
     @pytest.fixture(autouse=True)
     def _setup(self):
-        N = 4
-        self.matrixA = dq.destroy(N)
-        self.matrixB = dq.number(N)
-        self.sparseA = dq.to_sparse(self.matrixA)
-        self.sparseB = dq.to_sparse(self.matrixB)
+        N = 10
+        num_diags = 3
+        diags = jnp.arange(num_diags * N).reshape(num_diags, N)
+        offsets = tuple(range(-(num_diags // 2), (num_diags // 2) + 1))
```

Comment:
As we discussed, I think there's a mismatch between `num_diags` and the size of `offsets`. Once this is fixed, it's good to go for me.